site/_posts/2017-11-27-ray-0.3-release.markdown

```diff
@@ -0,0 +1,158 @@
+rllab's VisKit (or you can read the JSON format logs directly from the driver
+node):
+
+View the [documentation][2] and the [code][9].
+
+### RLlib integration with Ray.tune
+
+You can try out RLlib with Ray.tune with the following example.
+
+```bash
+cd ray/python/ray/rllib
+python train.py -f tuned_examples/cartpole-grid-search-example.yaml
+```
+
+The `tuned_examples` directory also contains pre-tuned hyperparameter
+configurations for common benchmark tasks such as Pong and Humanoid. View the
+[RLlib documentation][3].
+
+### Initial PyTorch support for RLlib
+
+A modern reinforcement learning library should work with multiple deep learning
+frameworks. As a step toward this goal, 0.3 adds support for PyTorch models for
+A3C in RLlib. You can try this out with the following A3C config.
+
+```bash
+cd ray/python/ray/rllib
+./train.py --run=A3C \
+           --env=PongDeterministic-v4 \
+           --config='{"use_pytorch": true, "num_workers": 8, "use_lstm": false, "model": {"grayscale": true, "zero_mean": false, "dim": 80, "channel_major": true}}'
+```
+
+## Distributed Actors
+
+Ray 0.3 adds support for [distributed actors][1], that is, the ability to have
+multiple callers invoke methods on the same actor. The actor’s creator may pass
+the actor handle as an argument to other tasks or to other actor methods. Here’s
+an example in which the driver creates an actor to log messages and passes an
+actor handle to other tasks:
+
+We've already found actor handles useful for implementing [**parameter
+server**][10] and [**streaming MapReduce**][11] applications.
+
+```python
+import ray
+
+ray.init()
+
+@ray.remote
+class Logger(object):
+    def __init__(self):
+        self.logs = []
+    def log(self, log_msg):
+        self.logs.append(log_msg)
+    def read_logs(self):
+        return self.logs
+
+@ray.remote
+def task(logger, task_index):
+    # Do some work.
+    logged = logger.log.remote('Task {} is done'.format(task_index))
+
+# Create an actor and get a handle to it.
+logger = Logger.remote()
+# Pass the actor handle to some tasks.
+tasks = [task.remote(logger, i) for i in range(10)]
+# Wait for the tasks to finish.
+ray.get(tasks)
+# Get the logs from the tasks.
+print(ray.get(logger.read_logs.remote()))
+```
+
+This feature is still under development, and full functionality is coming soon.
+Currently, the feature behaves as follows.
+
+- Execution order follows submission order, per handle. This means that Ray only
+  guarantees in-order execution between method calls from the same handle. No
+  ordering guarantees are currently provided between methods invoked from
+  different handles to the same actor.
+- Actor replay includes method calls submitted by the actor’s creator through
+  the original handle. If the actor is lost, tasks submitted through passed
+  handles may not re-execute when the actor is reconstructed.
+- Distributed actor replay currently does not support checkpointing.
```

Comment:
I would elaborate on checkpointing here. What exactly does it mean? That actor reconstruction will not use checkpointed state? Or it will not checkpoint during the process of reconstruction? Does the reader know about actor state checkpointing as a feature? This bullet could really be expanded into a couple of sentences.